testrunner-running/src/main/scala/edu/illinois/cs/statecapture/StateCapture.java

```diff
@@ -0,0 +1,755 @@
+                f.createNewFile();
+
+                FileWriter fw = new FileWriter(f.getAbsoluteFile(), append);
+                BufferedWriter w = new BufferedWriter(fw);
+                w.write(content);
+                w.close();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    Set<String> File2SetString(String path) {
+        File file = new File(path);
+        Set<String> keys = new HashSet<>();
+        try (BufferedReader br = new BufferedReader(new FileReader(file))) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                keys.add(line);
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return keys;
+    }
+
+    public void setup() {
+        subxmlFold = MainAgent.subxmlFold;
+        rootFold = MainAgent.rootFold;
+        diffFold = MainAgent.diffFold;
+        slug = MainAgent.slug;
+        reflectionFile = MainAgent.reflectionFold + "/0.txt";
+        xmlFileNum = countDirNums(subxmlFold);
+    }
+
+    public void fixing(String fieldName) throws IOException {
+        String subxml0 = subxmlFold + "/0xml";
+        try {
+            String path0 = subxml0 + "/" + fieldName + ".xml";
+            String state0 = readFile(path0);
+            String className = fieldName.substring(0, fieldName.lastIndexOf("."));
+            String subFieldName = fieldName.substring(fieldName.lastIndexOf(".")+1, fieldName.length());
+
+            Object ob_0;
+            XStream xstream = getXStreamInstance();
+
+            try {
+                Class c = Class.forName(className);
+                Field[] Flist = c.getDeclaredFields();
+                for (int i=0; i< Flist.length; i++) {
+                    if (Flist[i].getName().equals(subFieldName)) {
+                        try {
+                            Flist[i].setAccessible(true);
+                            Field modifiersField = Field.class.getDeclaredField("modifiers");
+                            modifiersField.setAccessible(true);
+                            modifiersField.setInt(Flist[i], Flist[i].getModifiers() & ~Modifier.FINAL);
+                        }
+                        catch (Exception e) {
+                            String outputPrivateError = fieldName + " reflectionError: " + e + "\n";
+                            Files.write(Paths.get(reflectionFile), outputPrivateError.getBytes(),
+                                    StandardOpenOption.APPEND);
+                        }
+                        try {
+                            ob_0 = Flist[i].get(null);
+                            boolean threadLocal = false;
+                            if (ob_0 instanceof ThreadLocal) {
+                                threadLocal = true;
+                            }
+                            try{
+                                if (Mockito.mockingDetails(ob_0).isMock()) {
```

Comment:
Let's have a comment that we want to try and directly invoke reset if we are dealing with a Mockito mock.